alibi_detect/od/sklearn/svm.py

```diff
@@ -0,0 +1,143 @@
+            Ignored by other kernels.
+        kernel_params
+            Additional parameters (keyword arguments) for kernel function passed as callable object.
+
+        Raises
+        ------
+        ValueError
+            If `n_components` is less than 1.
+        """
+        super().__init__()
+        self.n_components = n_components
+        self.kernel = kernel
+        self.sigma = sigma
+        self.kernel_params = kernel_params
+
+    def fit(  # type: ignore[override]
+        self,
+        x_ref: np.ndarray,
+        nu: float = 0.5,
+        tol: float = 1e-6,
+        max_iter: int = 1000,
+        verbose: int = 0,
+    ) -> Dict:
+        """Fit the SKLearn SVM model.
+
+        Parameters
+        ----------
+        x_ref
+            Training data.
+        nu
+            The proportion of the training data that should be considered outliers. Note that this does
+            not necessarily correspond to the false positive rate on test data, which is still defined when
+            calling the `infer_threshold` method.
+        tol
+            The decrease in loss required over the previous ``n_iter_no_change`` iterations in order to
+            continue optimizing.
+        max_iter
+            The maximum number of optimization steps.
+        verbose
+            Verbosity level during training. ``0`` is silent, ``1`` a progress bar.
+
+        Returns
+        -------
+        Dictionary with fit results. The dictionary contains the following keys:
+            - converged: `bool` indicating whether training converged.
+            - n_iter: number of iterations performed.
+        """
+        n_components = self.n_components if self.n_components is not None else len(x_ref)
+        self.nystroem = Nystroem(
+            kernel=self.kernel,
+            n_components=n_components,
+            gamma=1. / (2. * self.sigma ** 2) if self.sigma is not None else None,
+            kernel_params=self.kernel_params,
+        )
+        x_ref = self.nystroem.fit(x_ref).transform(x_ref)
+        self.gmm = SGDOneClassSVM(
+            tol=tol,
+            max_iter=max_iter,
+            verbose=verbose,
+            nu=nu
+        )
+        self.gmm = self.gmm.fit(x_ref)
+        self._set_fitted()
+        return {
+            'converged': self.gmm.n_iter_ < max_iter,
+            'n_iter': self.gmm.n_iter_,
+        }
+
+    def format_fit_kwargs(self, fit_kwargs: Dict) -> Dict:
+        """Format kwargs for `fit` method.
+
+        Parameters
+        ----------
+        kwargs
+            dictionary of Kwargs to format. See `fit` method for details.
+
+        Returns
+        -------
+        Formatted kwargs.
+        """
+        return dict(
+            nu=fit_kwargs.get('nu', 0.5),
+            tol=fit_kwargs.get('tol', 1e-3),
+            max_iter=fit_kwargs.get('max_iter', 1000),
+            verbose=fit_kwargs.get('verbose', 0),
```

Comment:
Slightly worried that defaults are defined in two places - here and in `fit`. Is there a better way of doing this?

Reply:
Good point, I've made an issue for [this](https://github.com/SeldonIO/alibi-detect/issues/819) as I use the same pattern in the `gmm` detector.

Reply:
👍🏻